Sources/client-sdk-swift/internal/PubsubClient.swift

```diff
@@ -81,19 +85,23 @@ class PubsubClient: PubsubClientProtocol {
         self.client = CacheClient_Pubsub_PubsubAsyncClient(channel: self.sharedChannel, interceptors: PubsubClientInterceptorFactory(credentialProvider: credentialProvider))
     }
     
-    func publish() async throws -> String {
+    func publish(
+        cacheName: String,
+        topicName: String,
+        value: String
+    ) async -> TopicPublishResponse {
         var request = CacheClient_Pubsub__PublishRequest()
-        request.cacheName = "cache-a-doodle"
-        request.topic = "bar"
-        request.value.text = "baz"
+        request.cacheName = cacheName
+        request.topic = topicName
+        request.value.text = value
         do {
             let result = try await self.client.publish(request)
-        } catch let myError as GRPCStatus {
-            throw grpcStatusToSdkError(grpcStatus: myError)
+            // Successful publish returns client_sdk_swift.CacheClient_Pubsub__Empty
+            self.logger.debug(msg: "Publish response: \(result)")
+            return TopicPublishSuccess()
         } catch {
-            print("Caught error while publishing:", error)
+            return TopicPublishError(error: grpcStatusToSdkError(grpcStatus: error as! GRPCStatus))
```

Comment:
I think we should specifically catch the `GRPCError` here: 
`} catch let err as GRPCStatus {`

That way you can just pass `err` to the `TopicPublishError` constructor already cast to the right type: 
`return TopicPublishError(error: grpcStatusToSdkError(grpcStatus: err))`

And we can add a final `} catch {` block to handle any errors that aren't coming from gRPC:
```
} catch {
    return TopicPublishError(error: UnknownError(message: "unknown error"))
}
```

Reply:
(The above won't work as-is because the error classes (including `UnknowError`) aren't public right now. I added a separate ticket for that, but there aren't that many of them and they're all in the same `Errors.swift` file, so if you wouldn't mind sweeping through and making all those SdkError subclasses public, please feel free!)